operators/src/machine_learning/onnx_util.rs

```diff
@@ -0,0 +1,318 @@
+    MetadataModelOutputShapeMismatch, MultipleInputsNotSupported, UnsupportedInOutMapping,
+    UnsupportedNumberOfOutputAttributes,
+};
+
+use super::{MachineLearningError, error::Ort};
+
+pub fn load_onnx_model_from_metadata(
+    ml_model_metadata: &MlModelMetadata,
+) -> Result<Session, MachineLearningError> {
+    ort::session::Session::builder()
+        .context(Ort)?
+        .commit_from_file(&ml_model_metadata.file_path)
+        .context(Ort)
+        .inspect_err(|e| {
+            tracing::debug!(
+                "Could not create ONNX session for {:?}. Error: {}",
+                ml_model_metadata.file_path.file_name(),
+                e
+            );
+        })
+}
+
+pub fn check_model_shape(
+    model_metadata: &MlModelMetadata,
+    tiling_shape: GridShape2D,
+) -> Result<(), MachineLearningError> {
+    check_model_input_shape_supported(model_metadata, tiling_shape)?;
+    check_model_output_shape_supported(model_metadata, tiling_shape)?;
+    check_input_output_mapping_supported(model_metadata)
+}
+
+pub fn check_model_input_shape_supported(
+    model_metadata: &MlModelMetadata,
+    tiling_shape: GridShape2D,
+) -> Result<(), MachineLearningError> {
+    // check that we can use the model input shape with the operator
+    ensure!(
+        model_metadata.input_is_single_pixel()
+            || model_metadata
+                .input_shape
+                .yx_matches_tile_shape(&tiling_shape),
+        InvalidInputPixelShape {
+            tensor_shape: model_metadata.input_shape,
+            tiling_shape
+        }
+    );
+
+    Ok(())
+}
+
+pub fn check_model_output_shape_supported(
+    model_metadata: &MlModelMetadata,
+    tiling_shape: GridShape2D,
+) -> Result<(), MachineLearningError> {
+    // check that we can use the model output shape with the operator
+    ensure!(
+        model_metadata.output_is_single_pixel()
+            || model_metadata
+                .output_shape
+                .yx_matches_tile_shape(&tiling_shape),
+        InvalidOutputPixelShape {
+            tensor_shape: model_metadata.output_shape,
+            tiling_shape
+        }
+    );
+
+    ensure!(
+        model_metadata.output_is_single_attribute(),
+        UnsupportedNumberOfOutputAttributes {
+            output_attributes: model_metadata.num_output_bands()
+        }
+    );
+
+    Ok(())
+}
+
+pub fn check_input_output_mapping_supported(
+    model_metadata: &MlModelMetadata,
+) -> Result<(), MachineLearningError> {
+    ensure!(
+        model_metadata.input_shape.axis_size_x() == model_metadata.output_shape.axis_size_x()
+            && model_metadata.input_shape.axis_size_y()
+                == model_metadata.output_shape.axis_size_y(),
+        UnsupportedInOutMapping {
+            in_shape: model_metadata.input_shape,
+            out_shape: model_metadata.output_shape
+        }
+    );
+
+    Ok(())
+}
+
+pub fn try_onnx_tensor_to_ml_tensorshape_3d(
```

Comment:
maybe test this method

Reply:
[7707c68](https://github.com/geo-engine/geoengine/pull/1000/commits/7707c68b2ee460f3454cd961e966523434cfadc5)